### ISSUE TITLE
[BUG] Can't import latest version of xgcm

### ISSUE THREAD

**Issue by navidcy** [xgcm/xgcm#262]
**What happened**:
I installed `xgcm` from commit b2fc97e and couldn't import it.
**What you expected to happen**:
I expected to be able to import.
**Minimal Complete Verifiable Example**:

```bash
python -m pip install --user git+https://github.com/xgcm/xgcm.git@b2fc97e29c6e262f69dc699d4c31dd2ad3ee53fc --upgrade

  Cloning https://github.com/xgcm/xgcm.git (to revision b2fc97e29c6e262f69dc699d4c31dd2ad3ee53fc) to /scratch/v45/nc3020/tmp/pip-req-build-94nh9e6b
  Running command git clone -q https://github.com/xgcm/xgcm.git /scratch/v45/nc3020/tmp/pip-req-build-94nh9e6b
Requirement already satisfied, skipping upgrade: xarray>=0.14.1 in /g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages (from xgcm==0.5.0+5.gb2fc97e) (0.16.1)
Requirement already satisfied, skipping upgrade: dask in /g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages (from xgcm==0.5.0+5.gb2fc97e) (2.28.0)
Requirement already satisfied, skipping upgrade: numpy in ./.local/lib/python3.7/site-packages (from xgcm==0.5.0+5.gb2fc97e) (1.19.1)
Requirement already satisfied, skipping upgrade: future in ./.local/lib/python3.7/site-packages (from xgcm==0.5.0+5.gb2fc97e) (0.18.2)
Requirement already satisfied, skipping upgrade: docrep<=0.2.7 in ./.local/lib/python3.7/site-packages (from xgcm==0.5.0+5.gb2fc97e) (0.2.7)
Requirement already satisfied, skipping upgrade: setuptools>=38.4 in /g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages (from xarray>=0.14.1->xgcm==0.5.0+5.gb2fc97e) (49.2.0.post20200712)
Requirement already satisfied, skipping upgrade: pandas>=0.25 in /g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages (from xarray>=0.14.1->xgcm==0.5.0+5.gb2fc97e) (1.1.2)
Requirement already satisfied, skipping upgrade: pyyaml in /g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages (from dask->xgcm==0.5.0+5.gb2fc97e) (5.3.1)
Requirement already satisfied, skipping upgrade: six in /g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages (from docrep<=0.2.7->xgcm==0.5.0+5.gb2fc97e) (1.15.0)
Requirement already satisfied, skipping upgrade: pytz>=2017.2 in /g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages (from pandas>=0.25->xarray>=0.14.1->xgcm==0.5.0+5.gb2fc97e) (2020.1)
Requirement already satisfied, skipping upgrade: python-dateutil>=2.7.3 in /g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages (from pandas>=0.25->xarray>=0.14.1->xgcm==0.5.0+5.gb2fc97e) (2.8.1)
Building wheels for collected packages: xgcm
  Building wheel for xgcm (setup.py) ... done
  Created wheel for xgcm: filename=xgcm-0.5.0+5.gb2fc97e-py3-none-any.whl size=60190 sha256=4901447ef2bb3b5bfae8f7c6ed399f70223a461406d9d33b0c4d4e72fafa9973
  Stored in directory: /home/552/nc3020/.cache/pip/wheels/31/e5/2a/2ebc2c994ecc59d2a757e2f93a388cddba68c78c374254da23
Successfully built xgcm
Installing collected packages: xgcm
Successfully installed xgcm-0.5.0+5.gb2fc97e
```
```python
Python 3.7.8 | packaged by conda-forge | (default, Jul 31 2020, 02:25:08)
[GCC 7.5.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import xgcm
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/552/nc3020/.local/lib/python3.7/site-packages/xgcm/__init__.py", line 6, in <module>
    from .grid import Grid, Axis
  File "/home/552/nc3020/.local/lib/python3.7/site-packages/xgcm/grid.py", line 19, in <module>
    from .transform import linear_interpolation, conservative_interpolation
  File "/home/552/nc3020/.local/lib/python3.7/site-packages/xgcm/transform.py", line 18, in <module>
    nopython=True,
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/npyufunc/decorators.py", line 179, in wrap
    guvec.add(fty)
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/npyufunc/ufuncbuilder.py", line 215, in add
    self.nb_func, targetoptions, sig)
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/npyufunc/ufuncbuilder.py", line 146, in _compile_element_wise_function
    cres = nb_func.compile(sig, **targetoptions)
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/npyufunc/ufuncbuilder.py", line 95, in compile
    return self._compile_core(sig, flags, locals)
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/npyufunc/ufuncbuilder.py", line 130, in _compile_core
    flags=flags, locals=locals)
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/compiler.py", line 551, in compile_extra
    return pipeline.compile_extra(func)
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/compiler.py", line 331, in compile_extra
    return self._compile_bytecode()
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/compiler.py", line 393, in _compile_bytecode
    return self._compile_core()
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/compiler.py", line 373, in _compile_core
    raise e
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/compiler.py", line 364, in _compile_core
    pm.run(self.state)
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/compiler_machinery.py", line 347, in run
    raise patched_exception
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/compiler_machinery.py", line 338, in run
    self._runPass(idx, pass_inst, state)
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/compiler_lock.py", line 32, in _acquire_compile_lock
    return func(*args, **kwargs)
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/compiler_machinery.py", line 302, in _runPass
    mutated |= check(pss.run_pass, internal_state)
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/compiler_machinery.py", line 275, in check
    mangled = func(compiler_state)
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/typed_passes.py", line 95, in run_pass
    raise_errors=self._raise_errors)
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/typed_passes.py", line 67, in type_inference_stage
    infer.propagate(raise_errors=raise_errors)
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-20.07/lib/python3.7/site-packages/numba/typeinfer.py", line 985, in propagate
    raise errors[0]
numba.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Use of unsupported NumPy function 'numpy.flip' or unsupported use of the function.

File ".local/lib/python3.7/site-packages/xgcm/transform.py", line 23:
def _interp_1d_linear(phi, theta, target_theta_levels, mask_edges, output):
    <source elided>
    if theta[-1] < theta[0]:
        theta = np.flip(theta)
        ^

[1] During: typing of get attribute at /home/552/nc3020/.local/lib/python3.7/site-packages/xgcm/transform.py (23)

File ".local/lib/python3.7/site-packages/xgcm/transform.py", line 23:
def _interp_1d_linear(phi, theta, target_theta_levels, mask_edges, output):
    <source elided>
    if theta[-1] < theta[0]:
        theta = np.flip(theta)
        ^
```

**Anything else we need to know?**:

**xgcm version and environment**:
xgcm 0.5.0+5.gb2fc97e
<details>

</details>

**Comment by navidcy** [comment]
same issue if I use the tag `v0.5.0`

**Comment by rabernat** [comment]
Thanks for the bug report @navidcy. This is probably a numba version issue. I think it should work if you upgrade numba.

We need to figure out how to handle this. Since numba is an optional dependency, we can't require a specific version. I would just replace
```python
theta = np.flip(theta)
```
with
```python
theta = theta[::-1]
```

Which should accomplish the exact same thing.

**Comment by jbusecke** [comment]
I'll work on a PR to fix both this and #261. Might not get it today but likely tomorrow.
